### PR TITLE
Added pre-commit linting as GitHub workflow

### PR DESCRIPTION
**Pull request by NyanKiyoshi**
Adds pre-commit checks.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by maarcingebala** [comment]
We could also drop Travis config in this PR since all checks are now run as GH actions.

**Review by maarcingebala — APPROVED** [review]

**Review by tomaszszymanski129 — APPROVED** [review]
LGTM :+1:

**Review by GrzegorzDerdak — APPROVED** [review]

**Review by fowczarek — COMMENTED** [review]
I'm not sure that it's good to add a new line at the end of the `.svg` file?

**Comment by fowczarek** [comment]
@NyanKiyoshi  This new checks still trigger `Codecov Report`?

**Comment by NyanKiyoshi** [comment]
@fowczarek yes, that's part of #5787 - svg I don't know but I prefer excluding them as they are generated files

File: templates/home/images/dashboard.svg [hunk 1]
```diff
@@ -1 +1 @@
.../><path d="M84.665 119.053l-5.04 2.52V74.131l5.04-2.52v47.442z" fill="#fff" stroke="#32335A" stroke-width=".7" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/><path d="M79.626 121.574l-37.311-18.656V55.575l37.31 18.655v47.344z" fill="#fff" stroke="#32335A" stroke-width=".75" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/><path d="M60.907 94.607l4.687-1.355v-5.694l-4.686-6.101-4.822 1.384v5.689l4.82 6.077z" fill="#fff" stroke="#32335A" stroke-width=".7"/><path d="M60.853 88.955v5.356" stroke="#32335A" stroke-width=".7"/><path d="M99.04 50.251L1 1.231v4.105L99.04 54.33V50.25z" fill="#EBF1F6" stroke="#32335A" stroke-width=".59" stroke-miterlimit="10"/><path d="M7.956 7.484c-.308-.385-.359-.846-.128-1.026.23-.18.667-.026.95.36.307.384.358.846.127 1.025-.23.205-.667.026-.949-.359zM5.212 6.099c-.308-.385-.36-.847-.128-1.026.23-.18.667-.026.949.359.308.385.359.847.128 1.026-.23.18-.667.026-.949-.359zM2.468 4.688c-.308-.385-.36-.846-.129-1.026.231-.18.667-.026.95.36.307.384.359.846.128 1.025-.231.205-.667.026-.95-.359z" fill="#fff" stroke="#28234A" stroke-width=".59" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/></svg>
```
Comment (fowczarek) [inline templates/home/images/dashboard.svg:1]: We should revert this change.

File: templates/home/images/logo.svg [hunk 2]
```diff
@@ -1 +1 @@
...2 2.02v1.168h-2.153c-5.837 0-7.826 1.768-7.826 5.367 0 3.535 1.793 5.24 5.478 5.24zM38.647 31.75h4.696V7.393l-4.696 4.656v19.7zM49.09 21.46v-.853c0-1.483.685-2.494 2.739-2.494 1.989 0 2.641.853 2.641 2.494v.853h-5.38zm2.733 10.542c5.413 0 6.848-2.525 7.174-5.05l-4.467-.663c-.196 1.515-.49 2.304-2.707 2.304-2.25 0-2.74-1.105-2.74-2.557v-1.389h9.914v-3.03c0-4.23-1.663-7.04-7.174-7.04-5.446 0-7.304 2.842-7.304 6.44v4.546c0 3.504 1.89 6.44 7.304 6.44zM67.703 32.002c5.38 0 7.598-2.746 7.598-6.44v-4.545c0-3.756-2.217-6.44-7.598-6.44-5.38 0-7.533 2.684-7.533 6.44v4.546c0 3.693 2.12 6.44 7.533 6.44zm-.002-3.661c-1.891 0-2.836-.979-2.836-2.557v-4.861c0-1.61.945-2.589 2.837-2.589 1.89 0 2.902.979 2.902 2.588v4.862c0 1.578-1.011 2.557-2.903 2.557zM83.688 19.395H81.17v12.256h-4.696V19.395l4.588-4.655h7.212l-4.587 4.655z"/></g><path fill-rule="evenodd" clip-rule="evenodd" d="M21.217 0l-6.18 6.44h15.267L36.484 0H21.218z" fill="#13BEBB"/><path fill-rule="evenodd" clip-rule="evenodd" d="M8.018 2.24a.602.602 0 01.425-.176h20.08a.601.601 0 01.425 1.028l-5.79 5.773a.602.602 0 01-.425.176H2.653a.602.602 0 01-.425-1.028l5.79-5.773zm.674 1.027l-4.584 4.57h18.376l4.583-4.57H8.692z" fill="#28234A"/></svg>
```
Comment (fowczarek) [inline templates/home/images/logo.svg:1]: We should revert this change.

File: templates/home/images/playground.svg [hunk 3]
```diff
@@ -1 +1 @@
...atrix(.94652 .46984 .00012 -.93987 60.296 83.05)" stroke="#247FFF" stroke-width=".5" d="M0-.25h.601"/><path transform="matrix(.94652 .46984 .00012 -.93987 61.19 83.494)" stroke="#247FFF" stroke-width=".5" d="M0-.25h.601"/><path transform="matrix(.94652 .46984 .00012 -.93987 53.459 75.662)" stroke="#247FFF" stroke-width=".5" d="M0-.25h.601"/><path transform="matrix(.94652 .46984 .00012 -.93987 51.886 73.486)" stroke="#247FFF" stroke-width=".5" d="M0-.25h.601"/><path transform="matrix(.94652 .46984 .00012 -.93987 62.034 83.913)" stroke="#247FFF" stroke-width=".5" d="M0-.25h.601"/><path d="M99.039 50.251L1 1.231v4.104L99.039 54.33v-4.078z" fill="#fff" stroke="#28234A" stroke-miterlimit="10" stroke-width=".59"/><path d="M7.956 7.484c-.308-.385-.359-.847-.128-1.026.23-.18.667-.026.949.36.308.384.36.846.128 1.025-.23.205-.667.026-.949-.359zM5.212 6.099c-.308-.385-.36-.847-.128-1.026.23-.18.667-.026.949.36.308.384.359.846.128 1.025s-.667.026-.949-.359zM2.468 4.688c-.308-.385-.36-.846-.129-1.026s.667-.026.95.36c.307.384.359.846.128 1.025-.231.205-.667.026-.95-.359z" fill="#fff" stroke="#33345A" stroke-linecap="round" stroke-linejoin="round" stroke-miterlimit="10" stroke-width=".59"/></svg>
```
Comment (fowczarek) [inline templates/home/images/playground.svg:1]: We should revert this change.

File: templates/home/images/splash.svg [hunk 4]
```diff
@@ -1 +1 @@
....07 1.852 1.198 5.339 1.852 11.44-1.307 3.159-1.635 6.864-3.487 10.786-5.557 11.984-6.21 26.256-13.618 32.684-16.778 4.14-2.07 5.448-5.011 5.775-7.408.217-1.199.108-2.288 0-3.051 0-.436-.001-.762-.11-.871z" fill="#fff" stroke="#28244A" stroke-width=".49"/><path d="M702.509 680.339v-21.681l73.541-36.716v21.681l-73.541 36.716zM727.894 658.004l26.039-12.965" stroke="#33345A" stroke-width="1.609" stroke-linecap="round" stroke-linejoin="round"/><path d="M701.311 626.627l17.759-8.716M701.311 638.719l41.619-20.591M723.427 615.731l30.833-15.362" stroke="#33345A" stroke-width=".981" stroke-linecap="round" stroke-linejoin="round"/><path d="M791.41 384.543l-104.373 53.276v273.243l104.373-53.276V384.543z" stroke="#28244A" stroke-width=".49"/><path d="M171.012 455.2l22.661 11.331 22.552-11.331-22.552-11.33-22.661 11.33zM171.008 477.753v-22.552l22.661 11.33v22.552l-22.661-11.33z" fill="#F2F8F9" stroke="#33345A" stroke-width=".804" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/><path d="M216.22 477.753l-22.553 11.33v-22.552l22.553-11.33v22.552z" fill="#F2F8F9" stroke="#33345A" stroke-width=".804" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/></svg>
```
Comment (fowczarek) [inline templates/home/images/splash.svg:1]: We should revert this change.

File: templates/home/images/storefront.svg [hunk 5]
```diff
@@ -1 +1 @@
...18v1.668a.755.755 0 00.365.697c.24.136.38.05.38-.27v-1.671l.728.418zM19.404 98.882l-.924 1.224-.003 1.085-.72-.418.003-1.078-.919-2.294.806.468.489 1.31.015.006.489-.744.764.441zM22.513 100.675l-.01 2.839-.687-.397-.886-2.036-.008 1.521-.682-.395.009-2.838.684.397.886 2.012.01-1.497.684.394zM24.134 101.54a2.756 2.756 0 011.307 2.247c0 .889-.575 1.164-1.316.738a2.78 2.78 0 01-1.306-2.253c0-.877.58-1.155 1.315-.732zm0 2.232c.297.172.593.071.593-.403a1.3 1.3 0 00-.593-1.076c-.296-.175-.592-.071-.592.397a1.29 1.29 0 00.592 1.082zM29.236 104.557l-.72 2.43-.72-.415-.43-1.985-.432 1.488-.747-.433-.726-3.263.726.418.397 2.092.45-1.603.661.382.445 2.125.391-1.642.705.406z" fill="#32335A"/><path transform="matrix(-.91537 -.52796 -.05874 .93803 36.731 87.705)" stroke="#28234A" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" d="M1-1h9.599"/><path transform="matrix(-.91537 -.52796 -.05874 .93803 24.38 80.582)" stroke="#28234A" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" d="M1-1h17.658"/><path transform="matrix(-.91537 -.52796 -.05874 .93803 18.742 82.105)" stroke="#28234A" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" d="M1-1h11.813"/></svg>
```
Comment (fowczarek) [inline templates/home/images/storefront.svg:1]: We should revert this change.

**Review by IKarbowiak — APPROVED** [review]

File: .travis.yml [hunk 6]
```diff
@@ -1,67 +0,0 @@
-dist: xenial
```
Comment (patrys) [inline .travis.yml:1]: Are we ready to drop the entire file? There are linters there that are waiting for #5786

**Review by patrys — CHANGES_REQUESTED** [review]

**Review by fowczarek — APPROVED** [review]

File: .travis.yml (hunk 6 above)
Comment (NyanKiyoshi) [reply .travis.yml:1]: Black, flake, mypy and isort are in pre-commit and migrations and GraphQL schema checks are also in it.

The question for that PR is whether we would want to have those checks in a separate workflow. The issue with pre-commit checks (migrations, GraphQL schema) is that we cannot tell pre-commit to only run them and no the others.

**Review by patrys — APPROVED** [review]

**Review by patrys — APPROVED** [review]
